### PR TITLE
refactor: Fix deprecation warnings when building against Qt 5.15

### LINKED ISSUE #36: Deprecation warnings when building against Qt 5.15
For a while there's been a stack of Qt related warnings when building against Qt 5.15.  Someone might want to take a stab at cleaning them up, assuming we can keep the code backwards compatible, without using too many hacks. There may be more than this, but these are all of the unique ones I could see:

```bash
  CXX      qt/libbitcoinqt_a-modaloverlay.o
qt/bitcoin.cpp:259:53: warning: 'QFlags' is deprecated: Use default constructor instead [-Wdeprecated-declarations]
    SplashScreen *splash = new SplashScreen(m_node, nullptr, networkStyle);
                                                    ^
/usr/local/Cellar/qt/5.15.0/lib/QtCore.framework/Headers/qflags.h:123:5: note: 'QFlags' has been explicitly marked deprecated here
    QT_DEPRECATED_X("Use default constructor instead") Q_DECL_CONSTEXPR inline QFlags(Zero) noexcept : i(0) {}
```

```bash
  CXX      qt/libbitcoinqt_a-peertablemodel.o
qt/bitcoingui.cpp:1308:29: warning: 'pixmap' is deprecated: Use the other overload which returns QPixmap by-value [-Wdeprecated-declarations]
        if (labelProxyIcon->pixmap() == nullptr) {
                            ^
/usr/local/Cellar/qt/5.15.0/lib/QtWidgets.framework/Headers/qlabel.h:77:5: note: 'pixmap' has been explicitly marked deprecated here
    QT_DEPRECATED_VERSION_X(5, 15, "Use the other overload which returns QPixmap by-value")
```

```bash
  CXX      qt/libbitcoinqt_a-coincontroltreewidget.o
qt/optionsmodel.cpp:222:59: warning: 'split' is deprecated: Use Qt::SplitBehavior variant instead [-Wdeprecated-declarations]
    QStringList ip_port = settings.value(name).toString().split(":", QString::SkipEmptyParts);
                                                          ^
/usr/local/Cellar/qt/5.15.0/lib/QtCore.framework/Headers/qstring.h:603:23: note: 'split' has been explicitly marked deprecated here
    Q_REQUIRED_RESULT QT_DEPRECATED_VERSION_X_5_15("Use Qt::SplitBehavior variant instead")
```

```bash
qt/transactionview.cpp:278:17: warning: 'QDateTime' is deprecated: Use QDate::startOfDay() [-Wdeprecated-declarations]
                QDateTime(current),
                ^
/usr/local/Cellar/qt/5.15.0/lib/QtCore.framework/Headers/qdatetime.h:296:5: note: 'QDateTime' has been explicitly marked deprecated here
    QT_DEPRECATED_X("Use QDate::startOfDay()") explicit QDateTime(const QDate &);
```

```bash
  CXX      qt/libbitcoinqt_a-moc_bitcoinaddressvalidator.o
qt/sendcoinsdialog.cpp:174:72: warning: 'buttonClicked' is deprecated: Use QButtonGroup::idClicked(int) instead [-Wdeprecated-declarations]
        connect(ui->groupFee, static_cast<void (QButtonGroup::*)(int)>(&QButtonGroup::buttonClicked), this, &SendCoinsDialog::updateFeeSectionControls);
                                                                       ^
/usr/local/Cellar/qt/5.15.0/lib/QtWidgets.framework/Headers/qbuttongroup.h:89:5: note: 'buttonClicked' has been explicitly marked deprecated here
    QT_DEPRECATED_VERSION_X_5_15("Use QButtonGroup::idClicked(int) instead")
```

### PR DESCRIPTION
**Pull request by hebasto**
[What's New in Qt 5.15](https://doc.qt.io/qt-5/whatsnew515.html#deprecated-modules):
> To help preparing for the transition to Qt 6, numerous classes and member functions that will be removed from Qt 6.0 have been marked as deprecated in the Qt 5.15 release.

Fixes #36

### CODE REVIEW

**Comment by promag** [comment]
Concept ACK.

**Comment by fanquake** [comment]
Concept ACK

**Comment by hebasto** [comment]
Rebased 354b2bd0568a62bd00c211d1286f62479da4820a -> 705c1f0648c72aa97e0ee699ff9a3da23fc9bd61 ([pr46.01](https://github.com/hebasto/gui/commits/pr46.01) -> [pr46.02](https://github.com/hebasto/gui/commits/pr46.02)) due to the conflict with #35.

**Comment by Sjors** [comment]
As mentioned in #36 I don't see any deprecation warnings on macOS with homebrew QT 5.15, but at least this change doesn't introduce new issues for me.

**Comment by hebasto** [comment]
@vasild 
As you mention these warnings in https://github.com/bitcoin/bitcoin/pull/19716#issuecomment-676290875, mind looking into this PR?

**Review by promag — COMMENTED** [review]
Tested ACK 705c1f0648c72aa97e0ee699ff9a3da23fc9bd61 on macos with Apple clang version 11.0.3 (clang-1103.0.32.62) and brew qt 5.15.1.

**Comment by jonasschnelli** [comment]
utACK 705c1f0648c72aa97e0ee699ff9a3da23fc9bd61
Added 0.22 milestone.

File: src/qt/guiutil.cpp [hunk 1]
```diff
@@ -920,4 +920,35 @@ void PopupMenu(QMenu* menu, const QPoint& point, QAction* at_action)
...
+bool HasPixmap(const QLabel* label)
+{
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+    return !label->pixmap(Qt::ReturnByValue).isNull();
+#else
+    return label->pixmap() != nullptr;
+#endif
+}
+
+QImage GetImage(const QLabel* label)
```
Comment (MarcoFalke) [inline src/qt/guiutil.cpp:941]: nit: Could pass by reference?
```suggestion
QImage GetImage(const QLabel& label)
```

**Review by MarcoFalke — COMMENTED** [review]
Concept aCK

**Comment by MarcoFalke** [comment]
btw, I am getting a ton of warnings like this one:

```
  CXX      qt/libbitcoinqt_a-bantablemodel.o
In file included from qt/bantablemodel.cpp:5:
In file included from ./qt/bantablemodel.h:12:
In file included from /usr/include/qt5/QtCore/QAbstractTableModel:1:
/usr/include/qt5/QtCore/qabstractitemmodel.h:170:5: warning: 'metaObject' overrides a member function but is not marked 'override' [-Wsuggest-override]
    Q_OBJECT
    ^
/usr/include/qt5/QtCore/qobjectdefs.h:175:32: note: expanded from macro 'Q_OBJECT'
    virtual const QMetaObject *metaObject() const; \
                               ^
/usr/include/qt5/QtCore/qobject.h:129:5: note: overridden virtual function is here
    Q_OBJECT
    ^
/usr/include/qt5/QtCore/qobjectdefs.h:175:32: note: expanded from macro 'Q_OBJECT'
    virtual const QMetaObject *metaObject() const; \
                               ^

**Comment by hebasto** [comment]
Compiling on macOS 11.0.1 Big Sur (20B29) + Qt 5.15.1 gives the following warnings now:
- a load of `-W#pragma-messages` from boost 1.74.0
```
/usr/local/include/boost/function_output_iterator.hpp:14:1: warning: This header is deprecated. Use <boost/iterator/function_output_iterator.hpp> instead. [-W#pragma-messages]
BOOST_HEADER_DEPRECATED("<boost/iterator/function_output_iterator.hpp>")
```

- three `-Wdeprecated-declarations` about deprecated `NSUserNotification`
```
warning: 'NSUserNotification' is deprecated: first deprecated in macOS 11.0 - All NSUserNotifications API should be replaced with UserNotifications.frameworks API [-Wdeprecated-declarations]
```

- and
```
qt/guiutil.cpp:70:37: warning: 'SystemLocaleShortDate' is deprecated: Use QLocale [-Wdeprecated-declarations]
    return date.date().toString(Qt::SystemLocaleShortDate) + QString(" ") + date.toString("hh:mm");
                                    ^
/usr/local/Cellar/qt/5.15.1/lib/QtCore.framework/Headers/qnamespace.h:1280:31: note: 'SystemLocaleShortDate' has been explicitly marked deprecated here
        SystemLocaleShortDate Q_DECL_ENUMERATOR_DEPRECATED_X("Use QLocale"),
                              ^
/usr/local/Cellar/qt/5.15.1/lib/QtCore.framework/Headers/qcompilerdetection.h:1161:45: note: expanded from macro 'Q_DECL_ENUMERATOR_DEPRECATED_X'
#  define Q_DECL_ENUMERATOR_DEPRECATED_X(x) Q_DECL_DEPRECATED_X(x)
                                            ^
/usr/local/Cellar/qt/5.15.1/lib/QtCore.framework/Headers/qcompilerdetection.h:675:55: note: expanded from macro 'Q_DECL_DEPRECATED_X'
#    define Q_DECL_DEPRECATED_X(text) attribute ((__deprecated__(text)))
                                                      ^
1 warning generated.
```


Not seening `-Wsuggest-override` warnings at all :man_shrugging:

**Comment by MarcoFalke** [comment]
Oh the override warnings happen with clang on linux. E.g fedora or debian:

https://cirrus-ci.com/task/6217749342978048?command=make#L10798

**Pull request merged**